test/MUXControlsTestApp/InfoBar.cs

```diff
@@ -0,0 +1,473 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Documents;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI;
+using System.Windows.Input;
+using Windows.Foundation;
+using Windows.UI.Composition;
+
+namespace MUXControlsTestApp
+{
+    public enum InfoBarCloseReason
+    {
+        CloseButton,
+        Programattic
+    }
+    public enum InfoBarSeverity
+    {
+        Critical,
+        Warning,
+        Informational,
+        Success,
+        Default,
+        None
+    }
+
+    public class InfoBarClosedEventArgs : EventArgs
+    {
+        public InfoBarCloseReason Reason
+        {
+            get; set;
+        }
+    }
+
+    public class InfoBarClosingEventArgs : EventArgs
+    {
+        public InfoBarCloseReason Reason
+        {
+            get; set;
+        }
+        public bool Cancel
+        {
+            get; set;
+        }
+
+    }
+    public class InfoBarEventArgs : EventArgs
+    {
+        public bool IsHandled
+        {
+            get; set;
+        }
+    }
+    public sealed class InfoBar : ContentControl
+    {
+        Button _actionButton;
+        Button _alternateCloseButton;
+        Button _closeButton;
+        Border _myBorder;
+
+        public event EventHandler<RoutedEventArgs> ActionButtonClick;
+        public event EventHandler<InfoBarEventArgs> CloseButtonClick;
+        public event TypedEventHandler<InfoBar, InfoBarClosedEventArgs> Closed;
+        public event TypedEventHandler<InfoBar, InfoBarClosingEventArgs> Closing;
+
+        private InfoBarCloseReason lastCloseReason = InfoBarCloseReason.Programattic;
+
+
+        public InfoBar()
+        {
+            this.DefaultStyleKey = typeof(InfoBar);
+        }
+
+
+        T GetTemplateChild<T>(string name) where T : DependencyObject
+        {
+            var child = GetTemplateChild(name) as T;
+            if (child == null)
+            {
+                throw new NullReferenceException(name);
+            }
+            return child;
+        }
+
+
+        protected override void OnApplyTemplate()
+        {
+            _alternateCloseButton = GetTemplateChild<Button>("AlternateCloseButton");
+            _closeButton = GetTemplateChild<Button>("CloseButton");
+            _actionButton = GetTemplateChild<Button>("ActionButton");
+            _myBorder = GetTemplateChild<Border>("Container");
```

Comment:
a name like container or root would be better.

Reply:
Do you mean replacing the name of _myBorder with something like root or container?